builtin/providers/azurerm/resource_arm_traffic_manager_endpoint.go

```diff
@@ -0,0 +1,272 @@
+				Optional: true,
+				Computed: true,
+			},
+
+			"weight": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validateAzureRMTrafficManagerEndpointWeight,
+			},
+
+			"priority": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validateAzureRMTrafficManagerEndpointPriority,
+			},
+
+			"endpoint_location": {
+				Type:     schema.TypeString,
+				Optional: true,
+				// when targeting an Azure resource the location of that resource will be set on the endpoint
+				Computed:  true,
+				StateFunc: azureRMNormalizeLocation,
+			},
+
+			"min_child_endpoints": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceArmTrafficManagerEndpointCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).trafficManagerEndpointsClient
+
+	log.Printf("[INFO] preparing arguments for ARM TrafficManager Endpoint creation.")
+
+	name := d.Get("name").(string)
+	endpointType := d.Get("type").(string)
+	profileName := d.Get("profile_name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	params := trafficmanager.Endpoint{
+		Name:       &name,
+		Type:       &endpointType,
+		Properties: getArmTrafficManagerEndpointProperties(d),
+	}
+
+	_, err := client.CreateOrUpdate(resGroup, profileName, path.Base(endpointType), name, params)
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(resGroup, profileName, path.Base(endpointType), name)
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read TrafficManager endpoint %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmTrafficManagerEndpointRead(d, meta)
+}
+
+func resourceArmTrafficManagerEndpointRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).trafficManagerEndpointsClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	endpointType := d.Get("type").(string)
+	profileName := id.Path["trafficManagerProfiles"]
+
+	// endpoint name is keyed by endpoint type in ARM ID
+	name := id.Path[path.Base(endpointType)]
+
+	resp, err := client.Get(resGroup, profileName, path.Base(endpointType), name)
+	if resp.StatusCode == http.StatusNotFound {
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("Error making Read request on TrafficManager Endpoint %s: %s", name, err)
+	}
+	endpoint := *resp.Properties
+
+	d.Set("name", resp.Name)
+	d.Set("endpoint_status", *endpoint.EndpointStatus)
+
+	if endpoint.TargetResourceID != nil {
+		d.Set("target_resource_id", *endpoint.TargetResourceID)
+	}
+
+	if endpoint.Target != nil {
+		d.Set("target", *endpoint.Target)
+	}
+
+	if endpoint.Weight != nil {
```

Comment:
Good to know, should the type conversions be removed too? e.g. `d.Set("weight", int(*endpoint.Weight))`